### DISCUSSION TITLE
boot from usb only with sd card inserted?

### DISCUSSION THREAD

**Discussion by arminposchmann** [Joshua-Riek/ubuntu-rockchip#48] [General]
First of all: Thanks to Joshua for your work, I am so glad I found this image for the orangepi5. It is so much better than the original ubuntu jammy image.
Sorry, that i have to ask this question, but i do not understand what i did wrong. Can someone please help? 
I installed the latest image 22.04 desktop onto an old ssd.
This did of course not boot from the ssd.
Then i prepared a sd card with this image. booted nicely from sd card.
took the rkspi_loader_sata.img from the original ubuntu jammy image and flashed it to /dev/mtdblock0.
Maybe this is the fault already, but i didn't find a  rkspi_loader.img 
Now the pi5 boots with sdcard inserted and the usb ssd connected.
When i look at the mounted devices /boot is on sda1, / is on sda2.
Without sd card inserted nothing happens when trying to boot.
I added "fdt_overlays=rk3588-ssd-sata.dtbo" to /boot/firmware/overlays.txt (on sda1) but this doesn't change anything.
Without sd card inserted the pi5 cannot boot.
rk3588-ssd-sata.dtbo is available under /boot/firmware/overlays.
Does anyone know what i can do to boot without sdcard inserted?

**Comment by Joshua-Riek** [comment]
I uploaded a new release about an hour ago with some bugfixes and rkspi_loader.img should be included now. But as for booting with a USB SSD, download the latest release to your sdcard and boot it up. Then flash rkspi_loader.img with the below command:
```
sudo dd if=/usr/share/orangepi/rkspi_loader.img of=/dev/mtdblock0
```
The command above may take up to 5 minutes to finish. After that you should be able to boot from a USB device without issue.

**Comment by Joshua-Riek** [comment]
Be sure to use the top blue USB port on your Orange Pi 5.

**Reply by arminposchmann** [reply]
No Change, sorry. 
First tried to boot from sdcard with v1.8 and the usb-disk having v1.7.
sudo dd if=/usr/share/orangepi/rkspi_loader.img of=/dev/mtdblock0
without sdcard: no boot attempt from usb-disk - with sdcard: boot and mount from sdcard (guess: wrong uuids on the old system)
then flashed the usb-disk to have v1.8 also 
without sdcard: no boot attempt from usb-disk - with sdcard boot and mount from usb-disk
echo "fdt_overlays=rk3588-ssd-sata.dtbo" | sudo tee /boot/firmware/overlays.txt
without sdcard: no boot attempt from usb-disk - with sdcard boot and mount from usb-disk

guess: As i can see that the usb adapter light does not even flicker without sdcard inserted it may be a hen and egg problem. 
Where can the information to use the ssd be stored? Should that information not go to the board itself, instead to usb-disk ?

Of course I am using the upper usb port, the lower seems to be usb2.0 only

**Reply by Joshua-Riek** [reply]
The bootloader is flashed to the onboard SPI chip, so the board can search for media such as a USB, NVME, or SD Card. But are you using a USB to SSD adapter or a regular USB stick?
